docs/index.rst

```diff
@@ -106,23 +106,79 @@ You can use pip and make to quickly set up a development environment::
+* Maintain Coding Standards
+    * Keep pep8 rules
+    * Follow the same stylistic and logical patterns used in the code
+        * All entity class names and class attributes have to be in the singular format
+        * All required entity attributes have to have `required=True` parameter
+        * It is preferable to use `alpha` data type for default string values for easier debug procedure
+        * In case any workaround is introduced, it is necessary to provide corresponding BZ ID directly into the code docstring
+        * All linting and formatting/style checks would be enforced by Travis-CI and PR would be considered broken until checks are passed successfully.
 * Adhere to typical commit guidelines:
-
-    * Commits should not cause NailGun to break.
+    * Commits should not cause NailGun’s unit test to fail. If it does, it will the responsibility of contributor to review those failures and fix them in the same PR's or raise another. The tracking of failures would be responsibility of contributor.
     * Commits should be small and coherent. One commit should address one issue.
     * Commits should have `good commit messages`_.
     * `Rebasing`_ is encouraged. Rebasing produces a much nicer commit history
       than merging.
+* To make the review process easy for all reviewers and anyone else interested in the new functionality, please provide some output making use of your changes. Having example of usage in docstring along with your code could really help others to build up on your code. You can add log from Python interactive shell or some tests results (from Robottelo / Foreman Ansible Modules) in PR message, or you can do something completely different - as long as it runs your code, it's fine!
+* If PR is applicable for many branches (e.g. master and one of '6.X.z' branches), specify that information in PR message
+* Unit tests
+    * Unit tests are compulsory
+    * Unit tests should cover all available actions, for the entity. For eg: Repository Sets, have enable, disable, list_available there should be unit tests exercising these actions.
+* When in doubt, ask on IRC. Join #robottelo on `freenode`_
+
+**Important to Note :**
+
+* Define Foreman Version labels in Nailgun
+    if possible, the contributor should set the right version.
+
+* All PRs should be raised along with Unit tests
+    The unit tests should be added while adding a new entity or modifying the existing entity or modifying and adding to the core of Nailgun.
+
+* Test results from upstream devel or from upstream nightly
+    The API call results are required to make the review process more firm. The PR author can provide results from any library that uses the contributed code by running the changes on upstream nightly or from his/her devel box. The interactive python shell output would be acceptable
```

Comment:
`The PR author can provide` -> `The PR author has to provide`

Reply:
This would change the meaning.

We require some kind of demonstration that Nailgun still works after introducing changes proposed in PR. First sentence says that. Next sentences show examples how this demonstration can be conducted - by running library that builds on Nailgun or by running Nailgun directly from Python shell. In fact, other ways are accepted as well - it's up to PR submitter to create convincing argument that her changes work.

If you change second sentence to "PR author has to provide", it will mean that PR must contain output from one of libraries that build on Nailgun; adding interactive shell log, or anything else, is optional. This is not what we actually do and this is not what we have agreed on. Current wording says precisely what we expect from contributors.

To put it other way - only first sentence is normative. Further sentences are descriptive, i.e. they describe how we usually meet this requirement. If you change "can" to "has" in second sentence, you move it from descriptive to normative.

Reply:
Yes, but only second sentence contains information about the person, so I am fine if we make change to the first sentence.
```
The API call results are required from PR author to make the review process more firm. He can provide results...
```

Reply:
We could argue that context of entire doc makes it clear that this is PR author responsibility, but yeah, I have no issue with stating it explicitly. Second proposal is OK for me :+1: